code/game/objects/items/weapons/storage/backpack.dm

```diff
@@ -563,6 +583,227 @@
 		new /obj/item/circular_saw(src)
 		new /obj/item/surgicaldrill(src)
 
+#define NANNY_MAX_VALUE 7
+#define NANNY_MIN_VALUE 6
+
+/obj/item/storage/backpack/duffel/magic_nanny_bag
+	name = "magic nanny bag"
+	desc = "Not to be confused with a magic granny bag. Zip it up to make it unable to be dropped while closed."
+	icon_state = "magic_nanny_bag"
+	item_state = "magic_nanny_bag"
+	max_w_class = WEIGHT_CLASS_HUGE
+	slot_flags = 0
+	storage_slots = 98 //Most that fits on your screen. Good luck getting that much in there.
+	max_combined_w_class = 256 //get your 8 bit magic bags here. Also it's wizard, at some point this many items will just make it crowded.
+	silent = TRUE
+	zip_time = 0
+	resistance_flags = FIRE_PROOF
+	open_icon_sprite = "magic_nanny_bag_open"
+	antidrop_on_zip = TRUE
+
+
+/obj/item/storage/backpack/duffel/magic_nanny_bag/populate_contents(attempts = 0)
```

Comment:
this proc could do with work. I think an one associative list, one for melee, one for wand, etc. would be better, with the key of the list being the "value" then randomly selecting from the list

Reply:
Moved to associated lists bar the things that have unequal probability, if anyone does know how to use pick weight with associated lists, I will do that

Reply:
As much as I don't like the switch statements, I don't have a good solution for this right now. It's also really messy/annoying dealing with 3 values (the weight, path _and_ value). I'm inclined to let this live, for now.